### ISSUE TITLE
[Feature Request] Add an Option to Log the Learning Rate in Progress Bar

### ISSUE THREAD

**Issue by ThomasElliot74** [Lightning-AI/lightning#14691]
## 🚀 Feature

See the learning rate at the progress bar output using the `LearningRateMonitor` class.

### Motivation

Seeing the learning rate as part of the progress bar will be a simple way to debug things without going into the logger.

### Pitch

Make things simpler :-).

### Alternatives

Do it manually.

______________________________________________________________________

#### If you enjoy Lightning, check out our other projects! ⚡

- [**Metrics**](https://github.com/Lightning-AI/metrics): Machine learning metrics for distributed, scalable PyTorch applications.

- [**Lite**](https://pytorch-lightning.readthedocs.io/en/latest/starter/lightning_lite.html): enables pure PyTorch users to scale their existing code on any kind of device while retaining full control over their own loops and optimization logic.

- [**Flash**](https://github.com/Lightning-AI/lightning-flash): The fastest way to get a Lightning baseline! A collection of tasks for fast prototyping, baselining, fine-tuning, and solving problems with deep learning.

- [**Bolts**](https://github.com/Lightning-AI/lightning-bolts): Pretrained SOTA Deep Learning models, callbacks, and more for research and production with PyTorch Lightning and PyTorch.

- [**Lightning Transformers**](https://github.com/Lightning-AI/lightning-transformers): Flexible interface for high-performance research using SOTA Transformers leveraging PyTorch Lightning, Transformers, and Hydra.

**Comment by carmocca** [comment]
You can do so with

```python
# assumes 1 optimizer with 1 param group
lr = self.optimizers().param_groups[0]["lr"]
self.log("lr", lr, prog_bar=True, on_step=True, on_epoch=False)
```

**Comment by ThomasElliot74** [comment]
I could also do the whole monitoring of the LR myself like that.
But if we're working to create a coherent way to do things, why not having it as an option in the class?

**Comment by carmocca** [comment]
Sorry, I see what you propose now. I think the `LearningRateMonitor` could be updated to use `pl_module.log` directly instead of manually sending the metrics to the logger.

Then, it could expose the `prog_bar` argument to resolve this issue.
Here's a quick idea of the potential changes:

```diff
@@ -87,13 +87,14 @@ class LearningRateMonitor(Callback):
 
     """
 
-    def __init__(self, logging_interval: Optional[str] = None, log_momentum: bool = False) -> None:
+    def __init__(self, logging_interval: Optional[str] = None, log_momentum: bool = False, prog_bar: bool = False) -> None:
         if logging_interval not in (None, "step", "epoch"):
             raise MisconfigurationException("logging_interval should be `step` or `epoch` or `None`.")
 
         self.logging_interval = logging_interval
         self.log_momentum = log_momentum
         self.lrs: Dict[str, List[float]] = {}
+        self.prog_bar = prog_bar
 
     def on_train_start(self, trainer: "pl.Trainer", *args: Any, **kwargs: Any) -> None:
         """Called before training, determines unique names for all lr schedulers in the case of multiple of the
@@ -147,26 +148,17 @@ class LearningRateMonitor(Callback):
         self.lrs = {name: [] for name in names_flatten}
         self.last_momentum_values = {name + "-momentum": None for name in names_flatten}
 
-    def on_train_batch_start(self, trainer: "pl.Trainer", *args: Any, **kwargs: Any) -> None:
-        if not trainer._logger_connector.should_update_logs:
-            return
-
+    def on_train_batch_start(self, trainer: "pl.Trainer", pl_module: "pl.LightningModule", *args: Any, **kwargs: Any) -> None:
         if self.logging_interval != "epoch":
             interval = "step" if self.logging_interval is None else "any"
             latest_stat = self._extract_stats(trainer, interval)
+            pl_module.log_dict(latest_stat, prog_bar=self.prog_bar)
 
-            if latest_stat:
-                for logger in trainer.loggers:
-                    logger.log_metrics(latest_stat, step=trainer.fit_loop.epoch_loop._batches_that_stepped)
-
-    def on_train_epoch_start(self, trainer: "pl.Trainer", *args: Any, **kwargs: Any) -> None:
+    def on_train_epoch_start(self, trainer: "pl.Trainer", pl_module: "pl.LightningModule", *args: Any, **kwargs: Any) -> None:
         if self.logging_interval != "step":
             interval = "epoch" if self.logging_interval is None else "any"
             latest_stat = self._extract_stats(trainer, interval)
-
-            if latest_stat:
-                for logger in trainer.loggers:
-                    logger.log_metrics(latest_stat, step=trainer.fit_loop.epoch_loop._batches_that_stepped)
+            pl_module.log_dict(latest_stat, prog_bar=self.prog_bar)
 
     def _extract_stats(self, trainer: "pl.Trainer", interval: str) -> Dict[str, float]:
         latest_stat = {}

```

Feel free to explore this direction in a PR if you have the bandwidth!
